xwiki-authenticator-trusted-api/src/main/java/org/xwiki/contrib/authentication/DynamicRoleConfiguration.java

```diff
@@ -0,0 +1,71 @@
+package org.xwiki.contrib.authentication;
+
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+/**
+ * Dynamic role configuration.
+ *
+ * @version $Id$
+ * @since 1.5.2
```

Comment:
Feels like what's this PR bring would deserve a 1.6.0. 1.5.2 would be only bugfixing or very minor things for me.